core/dummy_dao.py

```diff
@@ -0,0 +1,64 @@
+"""A dummy DAO implementation. Acts as a template for our
+future implementation.
+"""
+from slugify import slugify
+
+
+class FundDAO:
+    """A dummy interface to use instead of a database ORM."""
+
+    def __init__(self):
+        self.funds = {}
+
+    def get_one(self, fund_id):
+        return self.funds.get(fund_id)
+
+    def get_all(self):
+        return list(self.funds.values())
+
+    def create(self, data):
+        key = slugify(data["fund_name"])
+        data["fund_id"] = key
+        self.funds[key] = data
+
+    def load_dummy(self, fund_data):
+        for fund in fund_data:
+            self.create(fund)
+
+
+class RoundDAO:
+    """A dummy interface to use instead of a database ORM."""
+
+    def __init__(self):
+        self.rounds = {}
+
+    def get_one(self, fund_id, round_id):
+        all_round_list = list(self.rounds.values())
+        fund_round_list = [
+            round for round in all_round_list if round["fund_id"] == fund_id
+        ]
+        the_round = next(
+            (
+                round
+                for round in fund_round_list
+                if round["round_id"] == round_id
+            ),
+            None,
+        )
+        return the_round
+
+    def get_all(self, fund_id):
```

Comment:
This may arguably be better renamed to something like "search" or "filter", in that it returns a filtered subset of all rounds perhaps? Just for consistency and to avoid confusion when used in routes, may be worth having a separate get_all method that takes no other arguments and returns every round (mimicking the behaviour of get_all in the FundDAO) and rename this to "filter"?

Reply:
:100: makes complete sense! have renamed that method to say 'get_all_for_fund()' which hopefully is clearer that it's filtering rounds by the given fund!